search/search-domain/src/main/java/io/camunda/search/filter/UntypedOperation.java

```diff
@@ -46,4 +48,20 @@ public Object value() {
   public List<Object> values() {
     return values;
   }
+
+  private static void validateOrThrow(final Operator operator, final ValueTypeEnum type) {
+    switch (operator) {
+      case GREATER_THAN:
+      case GREATER_THAN_EQUALS:
+      case LOWER_THAN:
+      case LOWER_THAN_EQUALS:
+        if (type != ValueTypeEnum.LONG && type != ValueTypeEnum.DOUBLE) {
+          throw new IllegalArgumentException(
+              "Unsupported value type for number operator (<,<=,>,>=): " + type);
```

Comment:
❓Should we validate the `LIKE` & `STRING` as well?

Reply:
Yes, good idea! :)